NickvisionCavalier.Shared/Models/Renderer.cs

```diff
@@ -18,20 +30,342 @@ public void Draw(float[] sample, float width, float height)
+                }
+                if (_fill)
+                {
+                    path.LineTo(x + width, y);
+                    path.LineTo(x, y);
+                    path.Close();
+                }
+                break;
+            case DrawingDirection.BottomTop:
+                path.MoveTo(x, y + height * (1 - sample[0]) + (_fill ? 0 : _thickness / 2));
+                for (var i = 0; i < sample.Length - 1; i++)
+                {
+                    path.CubicTo(
+                        x + step * (i + 0.5f),
+                        y + height * (1 - sample[i]) + (_fill ? 0 : _thickness / 2),
+                        x + step * (i + 0.5f),
+                        y + height * (1 - sample[i+1]) + (_fill ? 0 : _thickness / 2),
+                        x + step * (i + 1),
+                        y + height * (1 - sample[i+1]) + (_fill ? 0 : _thickness / 2));
+                }
+                if (_fill)
+                {
+                    path.LineTo(x + width, y + height);
+                    path.LineTo(x, y + height);
+                    path.Close();
+                }
+                break;
+            case DrawingDirection.LeftRight:
+                path.MoveTo(x + width * sample[0] - (_fill ? 0 : _thickness / 2), y);
+                for (var i = 0; i < sample.Length - 1; i++)
+                {
+                    path.CubicTo(
+                        x + width * sample[i] - (_fill ? 0 : _thickness / 2),
+                        y + step * (i + 0.5f),
+                        x + width * sample[i+1] - (_fill ? 0 : _thickness / 2),
+                        y + step * (i + 0.5f),
+                        x + width * sample[i+1] - (_fill ? 0 : _thickness / 2),
+                        y + step * (i + 1));
+                }
+                if (_fill)
+                {
+                    path.LineTo(x, y + height);
+                    path.LineTo(x, y);
+                    path.Close();
+                }
+                break;
+            case DrawingDirection.RightLeft:
+                path.MoveTo(x + width * (1 - sample[0]) + (_fill ? 0 : _thickness / 2), y);
+                for (var i = 0; i < sample.Length - 1; i++)
+                {
+                    path.CubicTo(
+                        x + width * (1 - sample[i]) + (_fill ? 0 : _thickness / 2),
+                        y + step * (i + 0.5f),
+                        x + width * (1 - sample[i+1]) + (_fill ? 0 : _thickness / 2),
+                        y + step * (i + 0.5f),
+                        x + width * (1 - sample[i+1]) + (_fill ? 0 : _thickness / 2),
+                        y + step * (i + 1));
+                }
+                if (_fill)
+                {
+                    path.LineTo(x + width, y + height);
+                    path.LineTo(x + width, y);
+                    path.Close();
+                }
+                break;
+        }
+        Canvas.DrawPath(path, paint);
+        path.Dispose();
```

Comment:
If `SKPath` implements `IDisposable`, then you don't need to call `path.Dispose()`...All you need to do is add `using` in front of a disposable `var` so in this case: `using var path = new SKPath();` and C# will automatically call Dispose for you when needed